src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs

```diff
@@ -0,0 +1,232 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Pdb;
+using RegressionGames.RGLegacyInputUtility;
+using UnityEditor;
+using UnityEditor.Compilation;
+using UnityEngine;
+using Assembly = UnityEditor.Compilation.Assembly;
+
+namespace RegressionGames.Editor.RGLegacyInputUtility
+{
+    /*
+     * This class is responsible for hooking into the Unity build process and
+     * applying the wrapper input API instrumentation for the legacy input manager.
+     */
+    public class RGLegacyInputInstrumentation
+    {
+        [InitializeOnLoadMethod]
+        static void OnStartup()
+        {
+            // Safer to do any initial instrumentation within the editor update loop.
+            // Initial experiments with running directly from static initializers
+            // caused errors sometimes.
+            EditorApplication.update += SetUpHooks;
+        }
+
+        static void SetUpHooks()
+        {
+            if (EditorApplication.timeSinceStartup > 2.0)
```

Comment:
Timing things scare me, what is this trying to avoid ?

Reply:
The point of this is to do any initial instrumentation within the editor loop at some arbitrary point after the editor has started. When the package is first loaded, it iterates through any existing assemblies to check if they need to be instrumented (`InstrumentExistingAssemblies()`).

On my first attempt, I tried doing this immediately upon package load from within a static initializer, which caused the Unity runtime to refuse to execute some methods related to fetching the list of assemblies.

It is possible that this code would be perfectly fine without the timeSinceStartup check, but it seems safer with it and would still catch all cases: 1) Any assemblies that were compiled prior to this code running would be handled by `InstrumentExistingAssemblies()`; 2) Any assemblies compiled after this code runs would be checked by the `OnAssemblyCompiled` callback.

Reply:
@RG-nAmKcAz As I suspected, the timeSinceStartup check is not necessary here, so I've removed it and everything still works fine (verified on several different games).